### ISSUE TITLE
Slightly confusing interface during "Calculating Size"

### ISSUE THREAD

**Issue by pfrazee** [datproject/dat#544]
Really big files can take a few minutes to calculate the size. That looks like this:

![image](https://cloud.githubusercontent.com/assets/1270099/19365384/1bdfa57e-9158-11e6-84ca-6d3db7efa164.png)

After calculating the size, it will go to the progress bar, at which point things are clearer.

Based on the last line, it looks like it's actually serving, but it's not yet. Could we get rid of the "Watching for connections. Watching for updates" line during this phase?

**Comment by pfrazee** [comment]
(Shout out to @g-ray for bringing this to my attention)

**Comment by joehand** [comment]
It should be sharing right after the archive is created, before it even starts importing. Files are also added to the archive while it's still calculating the size - so you can share right away and people will get files before calculating is done. 

But agreed =), a confusing interface. We could hide waiting for connections unless there is actually connections at that point.

**Comment by pfrazee** [comment]
Maybe just putting the progress bar during calculating size would help.

**Comment by joehand** [comment]
@pfrazee finally understand what you meant here.

I fixed a bug that made it stay at calculating size after it was done calculating (at which point it'll show the adding progress bar). It was confusing because it'd stay at calculating until the first file was added. For large files that'd take a long time.

> Maybe just putting the progress bar during calculating size would help.

We can't show a progress bar because we don't how many files there are yet.

Hopefully this UI is better now because of the bug fix. Closing but if you see anything else or have idea for other UI around calculating, open it back up!
